test/org/sagebionetworks/bridge/models/subpopulations/ConsentSignatureTest.java

```diff
@@ -33,36 +33,39 @@ public void after() {
     }
     
     @Test
-    public void allDatesSuppressed() throws Exception {
+    public void allDatesFormattedCorrectly() throws Exception {
         ConsentSignature signature = new ConsentSignature.Builder()
             .withBirthdate("1970-01-01")
             .withName("Dave Test")
-            .withWithdrewOn(SIGNED_ON_TIMESTAMP)
-            .withConsentCreatedOn(SIGNED_ON_TIMESTAMP)
-            .withSignedOn(SIGNED_ON_TIMESTAMP).build();
+            .withWithdrewOn(WITHDREW_ON_TIMESTAMP.getMillis())
+            .withConsentCreatedOn(CONSENT_CREATED_ON_TIMESTAMP.getMillis())
+            .withSignedOn(SIGNED_ON_TIMESTAMP.getMillis()).build();
         
         String json = ConsentSignature.SIGNATURE_WRITER.writeValueAsString(signature);
         JsonNode node = BridgeObjectMapper.get().readTree(json);
-        assertNull(node.get("signedOn"));
         assertNull(node.get("consentCreatedOn"));
-        assertNull(node.get("withdrewOn"));
-        assertEquals("ConsentSignature", node.get("type").asText());
+        assertEquals(WITHDREW_ON_TIMESTAMP.toString(),
+                node.get("withdrewOn").textValue());
+        assertEquals(SIGNED_ON_TIMESTAMP.toString(), 
+                node.get("signedOn").textValue());
+        assertEquals("ConsentSignature", node.get("type").textValue());
         
         ConsentSignature deser = ConsentSignature.fromJSON(node);
         assertEquals("Dave Test", deser.getName());
         assertEquals("1970-01-01", deser.getBirthdate());
-        assertTrue(deser.getSignedOn() > 0L); // this is set in the builder
+        assertEquals(SIGNED_ON_TIMESTAMP.getMillis(), deser.getSignedOn()); // this is set in the builder
         assertEquals(0L, deser.getConsentCreatedOn());
```

Comment:
It's null, but that's wrong since I just added it to the SDK. Fixed here, will add a test to the integration tests.

Reply:
The ConsentSignature.fromJSON() method deserializes signatures sent from the client, and we don't allow the client to submit the timestamps. So they will not be deserialized. This looks like 0L for the consentCreatedOn timestamp because it's required, and null for withdrewOn because it can be null in the object we serialize and send to the client.

Having said that... I don't think we need the forJSON method, but that's more change in this PR to remove it.